keel/api/v1/auth/views.py

```diff
@@ -97,22 +90,125 @@ def login(self, request, format="json"):
         user = serializer.validated_data
-        token = JWTAuthentication.generate_token(user)
+
         try:
-            obj, created = CustomToken.objects.get_or_create(user=user, token=token)
+            token = JWTAuthentication.generate_token(user)
+            token_to_save = save_token(token)
+            obj, created = CustomToken.objects.get_or_create(user=user, token=token_to_save)
         except Exception as e:
-            logger.error('ERROR: AUTHENTICATION:UserViewset ' + str(e))
+            logger.error('ERROR: AUTHENTICATION:LoginViewset ' + str(e))
             response['message'] = str(e)
             response['status'] = 0
             return Response(response, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
         data = {
             "email" : obj.user.email,
-            "token" : obj.token['token'],
+            "token" : obj.token,
         }
         response["message"] = data
         return Response(response, status=status.HTTP_200_OK)
 
+class GeneratePasswordReset(GenericViewSet):
+    serializer_class = serializers.GenerateTokenSerializer
+    
+    def token(self, request):
+        response = {
+            "status" : 1,
+            "message" : ""
+        }
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        email = serializer.validated_data['email']
+        current_time = str(datetime.datetime.now().timestamp()).split(".")[0]
+        
+        try:
+            user = User.objects.get(email=email)
+            obj, created = PasswordResetToken.objects.get_or_create(user=user, reset_token=current_time)
+            context = {
+                'token' : current_time
+            }
+            subject = 'Password Reset'
+            html_content = get_template('password_reset_email.html').render(context)
+            # send email
+            emails = EmailNotification(subject, html_content, [email])
+            emails.send_email()
+        except Exception as e:
+            logger.error('ERROR: AUTHENTICATION:GeneratePasswordReset ' + str(e))
+            response['message'] = str(e)
+            response['status'] = 0
+            return Response(response, status=status.HTTP_400_BAD_REQUEST)
+        
+        response["message"] = "Password Reset Link sent successfully"
+        return Response(response)
+
+
+class ConfirmPasswordReset(GenericViewSet):
+    serializer_class = serializers.ConfirmResetTokenSerializer
+
+    def confirm_reset(self, request):
+        response = {
+            "status" : 1,
+            "message" : ""
+        }
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        validated_data = serializer.validated_data
+        reset_token = validated_data.get('reset_token', None)
+        password = validated_data['password']
+        try:
+            token = PasswordResetToken.objects.get(reset_token=reset_token)
```

Comment:
we also need to check if the token is not ore then 10 minutes old!!!!!...Also validate that this token belongs to the iuser it was sent to